taxi/forms.py

```diff
@@ -0,0 +1,59 @@
+import re
+
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+from django.core.exceptions import ValidationError
+
+from taxi.models import Driver, Car
+
+
+def validate_license_number(value):
+    if len(value) != 8:
+        raise ValidationError(
+            "License number must be exactly 8 characters long.",
+            code="invalid_length"
+        )
+    if not re.match(r"^[A-Z]{3}", value):
+        raise ValidationError(
+            "The first 3 characters must be uppercase letters",
+            code="invalid_first_three_characters"
+        )
+    if not re.search(r"\d{5}$", value):
+        raise ValidationError(
+            f"The last 5 characters must be numbers {value}",
+            code="invalid_last_five_characters"
+        )
+
+
+class CarCreateForm(forms.ModelForm):
+    drivers = forms.ModelMultipleChoiceField(
+        queryset=Driver.objects.all(),
+        widget=forms.CheckboxSelectMultiple(),
+        required=False,
+    )
+
+    class Meta:
+        model = Car
+        fields = "__all__"
+
+
+class DriverCreateForm(UserCreationForm):
+    license_number = forms.CharField(
+        validators=[validate_license_number],
+    )
+
+    class Meta(UserCreationForm.Meta):
+        model = Driver
+        fields = (
+            "username", "first_name", "last_name", "email", "license_number"
+        )
+
+
+class DriverLicenseUpdateForm(forms.ModelForm):
```

Comment:
So you saying this will not work for you?
```
class DriverLicenseUpdateForm(DriverCreateForm):

    class Meta:
        model = DriverCreateForm.Meta.model
        fields = ("license_number",)

Reply:
No, it doesn’t( 
<img width="516" alt="Screenshot 2024-09-24 at 15 47 02" src="https://github.com/user-attachments/assets/ed7ba324-3edc-481e-a413-efe538fd3a02">
<img width="548" alt="Screenshot 2024-09-24 at 15 45 35" src="https://github.com/user-attachments/assets/7099075b-af31-4672-b1a0-2268598b21a2">
I’ve tried it again, and it disrupts tests and the logic for updating the license number on the site page.